content/sensu-go/5.8/sensuctl/reference.md

```diff
@@ -386,6 +387,15 @@ See the [RBAC reference][21] for information about using access control with nam
 
 See the [RBAC reference][22] for information about local user management with sensuctl.
 
+## Pagination
+
+Sensuctl supports pagination for all `list` commands using the `--chunk-size` flag.
+For example, the following command returns only the first 20 events.
```

Comment:
@ccressent Is there anything else we can say here about the ordering of the objects? Is there a way to see the next X number of objects after the initial command?

Reply:
Sorry Alex, I must have communicated poorly about how this `--chunk-size` flag works.

In your example, `sensuctl event list --chunk-size 20` would retrieve _all_ the events. In fact, purely in term of output, there should never be a difference between `sensuctl x list` and `sensuctl x list --chunk-size y`!

What the flag does is this: instead of making a single query to the API to get all the events in one go, it instead makes a series of smaller queries to retrieve all the events in smaller chunks. For example, if there are 1000 events to be retrieved, and `chunk-size` is 20, `sensuctl` will issue 50 queries to retrieve all the events and then display them.

One might ask: why is that useful? That's because trying to retrieve big sets of objects from an API can lead to issues, such as the query timing out. If instead the client issues many smaller queries and then reassemble the response, most of those problems disappear.